controllers/directoryservice_controller.go

```diff
@@ -121,7 +128,7 @@ func (r *DirectoryServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		// temporary loop due to MCS limitation with statefulsets where we need a service per pod.
 		for i = 0; i < *ds.Spec.Replicas; i++ {
 			id := strconv.Itoa(int(i))
```

Comment:
We can take out the service hack now that MCS supports headless services?

Reply:
Yes, the new MCS support has a slightly different format. It includes the hub membership name in the URL.  So I need to update the dockerfile first. Then I can revert this.

Reply:
I have removed the service hack now and the extra replication service controller.